server/fly.io/Dockerfile

```
@@ -0,0 +1,26 @@
+FROM rust:1-buster as builder
+
+WORKDIR /usr/src/app
+COPY . .
+# Will build and cache the binary and dependent crates in release mode
+RUN --mount=type=cache,target=/usr/local/cargo,from=rust:latest,source=/usr/local/cargo \
+    --mount=type=cache,target=target \
+    cargo build --release && mv ./target/release/cargo-quickinstall-stats-server ./cargo-quickinstall-stats-server
```

I think compiling for musl makes more sense?

With musl and everything statically linked, we don't have to use this mounting from rust:latest and can use alpine, or even a scratch as base image

```suggestion
    cargo build --release --target x86_64-unknown-linux-musl && mv ./target/release/cargo-quickinstall-stats-server ./cargo-quickinstall-stats-server
```

---

Hrm. After fighting with debian for a bit, I might be coming round to your way of thinking. I'm going to plug away at it after lunch for a bit and see where I get to.

---

turns out I got it working with debian again. I'm going to backlog the problem of reacting to CVEs, I think.

---

Well, if you use vendored openssl, then whenever a new openssl is published, a new openssl-src will be published, and dependabot is able to create a PR to update them.

With auto deployment in GHA, it'd be a smooth process.

---

backlogged CVE stuff as (https://github.com/cargo-bins/cargo-quickinstall/issues/277) for now